### PR TITLE
Feat/shared image gallery update

### PR DESCRIPTION
**Pull request by joeybdub**
# [Issue-id](https://github.com/aztfmod/terraform-azurerm-caf/issues/ISSUE-ID-GOES-HERE)

## PR Checklist

---

- [ ] I have added example(s) inside the [./examples/] folder 
- [ ] I have added the example(s) to the integration test list for [normal (~30 minutes)](./workflows/standalone-scenarios.json) or [long runner >30 minutes](./workflows/standalone-scenarios-longrunners.json)
- [x ] I have checked the [coding conventions as per the wiki](https://github.com/aztfmod/terraform-azurerm-caf/wiki)
- [ x] I have checked to ensure there aren't other open Pull Requests for the same update/change?

## Description

Adding support for [hyper_v_generation] (https://registry.terraform.io/providers/hashicorp/azurerm/2.88.1/docs/resources/shared_image#hyper_v_generation)

## Does this introduce a breaking change

- [ ] YES
- [ x] NO

### CODE REVIEW

**Review by LaurentLesle — APPROVED** [review]
LGTM. Thanks for the PR

File: modules/shared_image_gallery/image_definitions/Image_definitions.tf [hunk 1]
```diff
@@ -15,6 +15,7 @@ resource "azurerm_shared_image" "image" {
   resource_group_name = var.resource_group_name
   location            = var.location
   os_type             = var.settings.os_type
+  hyper_v_generation  = var.hyper_v_generation
```
Comment (arnaudlh) [inline modules/shared_image_gallery/image_definitions/Image_definitions.tf:18]:
```suggestion
  hyper_v_generation  = try(var.settings.hyper_v_generation, null)
```
I would suggest we catch it under the settings object and we put it as optional as it is in the provider: https://registry.terraform.io/providers/hashicorp/azurerm/2.99.0/docs/resources/shared_image#hyper_v_generation

File: modules/shared_image_gallery/image_definitions/variables.tf [hunk 2]
```diff
@@ -6,4 +6,5 @@ variable "client_config" {}
 variable "global_settings" {}
 variable "settings" {}
 variable "base_tags" {}
-variable "gallery_name" {}
\ No newline at end of file
+variable "gallery_name" {}
+variable "hyper_v_generation" {}
```
Comment (arnaudlh) [inline modules/shared_image_gallery/image_definitions/variables.tf:10]:
```suggestion
```
Under "settings" object config.

**Review by arnaudlh — CHANGES_REQUESTED** [review]
Thanks for your PR! Please review and update if agree!

**Comment by joeybdub** [comment]
I have update the code as requested @arnaudlh @LaurentLesle

**Review by arnaudlh — APPROVED** [review]
LGTM

**Comment by arnaudlh** [comment]
thanks for your contribution @joeybdub!

**Comment by joeybdub** [comment]
Thanks and I hope to have some more in the near future!

**Pull request merged**